provider/pkg/provider/client/client.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2016-2024, Pulumi Corporation.
+
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudcontrol"
+	"github.com/aws/aws-sdk-go-v2/service/cloudcontrol/types"
+	"github.com/mattbaird/jsonpatch"
+	"github.com/pkg/errors"
+	"github.com/pulumi/pulumi-aws-native/provider/pkg/schema"
+)
+
+type Client interface {
```

Comment:
The name "Client" doesn't say much because there are many clients. I'd be more specific, but it's subjective, of course.

Reply:
@danielrbradley Are you okay to check this to CloudControlClient in your branch? I'd prefer that name to Client too.

Reply:
Sure - I agree that name would be better